osutils.pm

```diff
@@ -100,7 +100,7 @@ sub _run {
     $p->channels(0)->quirkiness(1)->internal_pipes(0)->separate_err(0)->start;
     $p->on(stop => sub {
             while (defined(my $line = $p->getline)) {
-                $out .= $line;
+                chomp($out .= $line);
```

Comment:
Just curious, do we really need the output flatten to one line?

Reply:
well, every "line" should just be that, one line, isn't it?

Reply:
I think, output like
```
a
b
c
```
become
`abc`
instead of only stripping the last `\n`. If this is what you won't, ok.

Reply:
Not according to my test:

```
  DB<1> $foo = "foo\nbar\nfoo\n";

  DB<2> print($foo);
foo
bar
foo

  DB<3> print(chomp($foo));
1
  DB<4> print($foo);
foo
bar
foo
  DB<5> 
```

Reply:
hm, because I also call chomp around the $out, could be. I should crosscheck this then.

Reply:
I agree what you are tested is what I would expect. But I think we have a different case, more like this:
```
open(my $f, '<', '/etc/os-release') || die("ERROR on read");
my $out = "";
while (my $line = readline($f)) {
        chomp($out .= $line);
}
print $out;
close($f);
```

Reply:
Maybe I have a misunderstanding from that `$p->getline` and it act different!?

Reply:
@cfconrad I get your point - my example is not like this code. Then the code looks wrong, indeed.

Reply:
any suggestion what I should do instead?

Reply:
You could `chomp($out);` before return.

Reply:
yes, of course! Thanks :)